optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStreamTest.java

```diff
@@ -404,6 +404,16 @@ public void join_mixedEqualsAndFiltering() {
     // If (not) exists
     // ************************************************************************
 
+    @TestTemplate
+    public void ifExists_unknownClass() {
+        assumeDrools();
+        assertThatThrownBy(() -> buildScoreDirector((factory) -> {
+            return factory.fromUniquePair(TestdataLavishEntity.class)
+                    .ifExists(Integer.class)
+                    .penalize(TEST_CONSTRAINT_NAME, SimpleScore.ONE);
+        })).isInstanceOf(IllegalArgumentException.class);
```

Comment:
If the code throws the `IllegalArgumentException` in a different situation, the test won't notice. The `IAE` is one of the most used exceptions in the optaplanner codebase.
Suggestion: add a check for a specific substring of the exception message.